config.inc.php

```diff
@@ -8,12 +8,27 @@
 $hosts = isset($_ENV['MYSQL_HOST']) ? $_ENV['MYSQL_HOST'] : 'mysql';
 foreach (explode(',', $hosts) as $index => $host) {
 	$config = &$cfg['Servers'][$index + 1];
-	$host   = trim($host);
+
+	// split into host[:port], user and pass
+	$parts = explode(";", $host);
```

Comment:
I'd suggest following as more readable approach:
```php
list($host, $user, $password) = explode(';', $host, 3) + [1 => '', 2 => ''];
```

It will also allow potential `;` symbol in password (while `,` will not be supported anyway).

You'll not need tricky `array_shift()` afterwards.